### ISSUE TITLE
Image Widgets - InvalidAuthenticityToken after logging in/out

### ISSUE THREAD

**Issue by ives1227** [projectblacklight/spotlight#2179]
When using any of the image widgets, if the user signs in or out and then clicks on one of the images in the widget, it throws an InvalidAuthenticityToken exception. I have been able to replicate this with a vanilla version of 2.7.0 version of Spotlight (first noticed in 2.3.0).  I've reproduced on FF, Chrome, and Safari.

Steps to reproduce:
1. Login to Spotlight and create or navigate to an exhibit
2. Create a page with an image widget such as Item Row and add a few images to it.
3. Make sure the exhibit and widget page are both published to be able to access when logged out.
4. Navigate to the page with the widget and click on one of the images.
5. Logout of Spotlight
6. Navigate back to the page with the widget and click on one of the images.  This should throw the exception.

This also works in the opposite direction.  For example, if you are logged in and then logout and then attempt to access the image via the image widget, it throws the same exception.

Clearing the cache does resolve the problem.

**Comment by jkeck** [comment]
@ives1227 does this only occur when using the browser's back button to navigate back to the page in step 6?

**Comment by ives1227** [comment]
No, I can even refresh the widget page and still get the error when I click on the image.  I created a screen recording here: https://youtu.be/H68nIePj6Og  NOTE: This demo not using the vanilla instance since I have to run that on my local environment and it was painfully slow when trying to create the screen recording on top of running a local instance.  The same problem occurred on the vanilla instance.  I created a feature page called 'widget example', added an Image Row, and followed the steps above.

**Comment by jkeck** [comment]
@ives1227 okay, so I've taken the following steps on a fresh install of 2.7.0.

* Created a new exhibit
* Added an Item Row widget to the home page (w/ two images).
* Published Exhibit
* Logged out
* Went to exhibit home page and clicked on an image
* Received the record that I expect

Am I missing something in the steps to reproduce this particular bug?

![item-row](https://user-images.githubusercontent.com/96776/60863566-088c5a80-a1d6-11e9-96eb-8b7f53af9da2.gif)

**Comment by ives1227** [comment]
@jkeck  this is on a feature page so perhaps that should be a step - try creating a feature page and then add an image row. I was not able to replicate this on the home page of an exhibit when I tried.

**Comment by jkeck** [comment]
@ives1227, ah, odd, I assumed they would behave the same.  I've been able to reproduce this error and believe it has to do w/ passing a search_id (to try and track search history like normal search results do) on the feature page where we aren't on the home page.

I will try to look further when I have time, but if that helps you or anybody else sleuth this out, please feel free to submit a fix.

**Comment by mejackreed** [comment]
I'm not sure if this is related, but we dug into this a bit in ArcLight and found out some things. Depending on the location of an applications rails-ujs and jquery requires, there can be some odd things happening with CSRF tokens. See https://github.com/projectblacklight/blacklight/pull/2171 for the generator update.

And this fix worked in a previously generated ArcLight application.

```diff
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
 //
+//= require jquery3
 //= require rails-ujs
 //= require activestorage
 //= require turbolinks
 //
 // Required by Blacklight
-//= require jquery
 //= require popper
 // Twitter Typeahead for autocomplete
```

@ives1227 Not sure if that will work for you, but wanted to call this out.
